cornellh4i/clean-the-world
# Check claims function

## Overview

Closes #10 : Refactored the createUser() Firebase function to give new users a custom dataEntry claim.

## Changes Made

Created a new customClaims object and set dataEntry to true for all newly created users. Also created a function to check whether a user's dataEntry claim is set to true or not, and a route that calls this function.

## Test Coverage

Created a route that takes in a user ID as a a parameter and checks whether that user has the dataEntry claim. Tested on new users after refactoring (dataEntry = true) and old users (dataEntry = false).

1. `yarn build` and `yarn start`
2. Create a new user at `localhost:3000/users/create/<phone number>`
3. Get their user ID (uid) via console or Firebase console.
4. Verify their dataEntry claim at `localhost:3000/users/check/<uid>`
5. `true` printed to console if that user has dataEntry, `false` if no dataEntry claim, and error if no user found.
